doc/en/cache.txt

```diff
@@ -0,0 +1,256 @@
+failures will be executed first (as can be seen from the series of ``FF`` and
+dots)::
+
+    $ py.test --ff
+    ============================= test session starts ==============================
+    platform linux2 -- Python 2.7.3 -- pytest-2.3.5
+    run-last-failure: rerun last 2 failures first
+    plugins: cache
+    collected 50 items
+
+    test_50.py FF................................................
+
+    =================================== FAILURES ===================================
+    _________________________________ test_num[17] _________________________________
+
+    i = 17
+
+        @pytest.mark.parametrize("i", range(50))
+        def test_num(i):
+            if i in (17,25):
+    >          pytest.fail("bad luck")
+    E          Failed: bad luck
+
+    test_50.py:6: Failed
+    _________________________________ test_num[25] _________________________________
+
+    i = 25
+
+        @pytest.mark.parametrize("i", range(50))
+        def test_num(i):
+            if i in (17,25):
+    >          pytest.fail("bad luck")
+    E          Failed: bad luck
+
+    test_50.py:6: Failed
+    ===================== 2 failed, 48 passed in 0.07 seconds ======================
+
+.. _`config.cache`:
+
+The new config.cache object
+--------------------------------
+
+.. regendoc:wipe
+
+Plugins or conftest.py support code can get a cached value
+using the pytest ``config`` object.  Here is a basic example
+plugin which implements a `funcarg <http://pytest.org/latest/funcargs.html>`_
+which re-uses previously created state across py.test invocations::
+
+    # content of test_caching.py
+    import time
+
+    def pytest_funcarg__mydata(request):
+        val = request.config.cache.get("example/value", None)
+        if val is None:
+            time.sleep(9*0.6) # expensive computation :)
+            val = 42
+            request.config.cache.set("example/value", val)
+        return val
+
+    def test_function(mydata):
+        assert mydata == 23
+
+If you run this command once, it will take a while because
+of the sleep::
+
+    $ py.test -q
+    F
+    =================================== FAILURES ===================================
+    ________________________________ test_function _________________________________
+
+    mydata = 42
+
+        def test_function(mydata):
+    >       assert mydata == 23
+    E       assert 42 == 23
+
+    test_caching.py:12: AssertionError
+
+If you run it a second time the value will be retrieved from
+the cache and this will be quick::
+
+    $ py.test -q
+    F
+    =================================== FAILURES ===================================
+    ________________________________ test_function _________________________________
+
+    mydata = 42
+
+        def test_function(mydata):
+    >       assert mydata == 23
+    E       assert 42 == 23
+
+    test_caching.py:12: AssertionError
+
+Consult the `pytest-cache API <http://packages.python.org/pytest-cache/api.html>`_
```

Comment:
This link still points to the external pytest-cache... anyway, I can't find the api docs here in the PR...